pkg/dockerregistry/server/tagservice.go

```diff
@@ -150,39 +152,38 @@ func (t tagService) Lookup(ctx context.Context, desc distribution.Descriptor) ([
 }
 
 func (t tagService) Tag(ctx context.Context, tag string, dgst distribution.Descriptor) error {
-	imageStream, err := t.repo.imageStreamGetter.get()
-	if err != nil {
-		context.GetLogger(ctx).Errorf("error retrieving ImageStream %s/%s: %v", t.repo.namespace, t.repo.name, err)
-		return distribution.ErrRepositoryUnknown{Name: t.repo.Named().Name()}
-	}
+	dockerImageReference := fmt.Sprintf("%s/%s/%s@%s", t.repo.config.registryAddr, t.repo.namespace, t.repo.name, dgst.Digest.String())
 
-	image, err := t.repo.registryOSClient.Images().Get(dgst.Digest.String(), metav1.GetOptions{})
-	if err != nil {
-		context.GetLogger(ctx).Errorf("unable to get image: %s", dgst.Digest.String())
-		return err
-	}
-	image.SetResourceVersion("")
-
-	if !t.repo.config.pullthrough && !isImageManaged(image) {
-		return distribution.ErrRepositoryUnknown{Name: t.repo.Named().Name()}
-	}
-
-	ism := imageapiv1.ImageStreamMapping{
+	isi := imageapiv1.ImageStreamImport{
```

Comment:
How will the image stream import work if the image isn't tagged in any image stream yet?

Reply:
The image is imported by the digest.

Reply:
The imagestreamimport requires a source repository name. What if there is no corresponding image stream yet?